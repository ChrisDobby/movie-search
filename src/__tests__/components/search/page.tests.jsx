import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import SearchPage from '../../../components/search/page';

describe('search page', () => {
    const mockConfig = ({
        baseImageUrl: 'http://www.google.com',
    });

    const mockSearch = [{}];

    const successfulActions = {
        search: jest.fn(() => Promise.resolve(mockSearch)),
    };

    const failingActions = {
        search: jest.fn(() => Promise.reject()),
    };

    it('should set state from succesful search action', () => {
        const page = shallow(<SearchPage actions={successfulActions} config={mockConfig} />);
        page.setState({
            configuration: { config: mockConfig },
        });

        page.instance().movieSearch('paddington');

        return new Promise(resolve => setImmediate(resolve))
            .then(() => {
                expect(page.state().search.searching).toBeFalsy();
                expect(page.state().search.searchError).toBeFalsy();
                expect(page.state().search.movies).toEqual(mockSearch);
            });
    });

    it('should set state from failed search action', () => {
        const page = shallow(<SearchPage actions={failingActions} config={mockConfig} />);
        page.setState({
            configuration: { config: mockConfig },
        });

        page.instance().movieSearch('paddington');

        return new Promise(resolve => setImmediate(resolve))
            .then(() => {
                expect(page.state().search.searching).toBeFalsy();
                expect(page.state().search.searchError).toBeTruthy();
                expect(page.state().search.movies).toEqual([]);
            });
    });

    it('should render correctly when searching', () => {
        const page = renderer.create(<SearchPage actions={successfulActions} config={mockConfig} />);
        page.getInstance().setState({
            search: {
                searching: true,
                searchError: false,
                movies: [],
            },
        });

        expect(page.toJSON()).toMatchSnapshot();
    });

    it('should render correctly when search failed', () => {
        const page = renderer.create(<SearchPage actions={successfulActions} config={mockConfig} />);
        page.getInstance().setState({
            search: {
                searching: false,
                searchError: true,
                movies: [],
            },
        });

        expect(page.toJSON()).toMatchSnapshot();
    });
});
