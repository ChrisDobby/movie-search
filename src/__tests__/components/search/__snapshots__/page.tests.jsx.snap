// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search page should render correctly when search failed 1`] = `
<div>
  <div
    className="alert alert-danger"
    role="alert"
  >
    There was an error searching.  Please try again.
  </div>
  <div
    className="progress"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="100"
      className="progress-bar progress-bar-striped progress-bar-animated"
      role="progressbar"
      style={
        Object {
          "width": "75%",
        }
      }
    />
  </div>
</div>
`;

exports[`search page should render correctly when searching 1`] = `
<div>
  <div
    className="progress"
  >
    <div
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="100"
      className="progress-bar progress-bar-striped progress-bar-animated"
      role="progressbar"
      style={
        Object {
          "width": "75%",
        }
      }
    />
  </div>
</div>
`;
